Fix: Support both the new Activity Log v2 links as well as the previous one in the Overview page.

andrea-sdl opened the pull request:
## Proposed Changes

* Adds support for both the `activity-log/<site>` link as well as the `/v2/sites/<site>/logs/activity` depending on where you are looking at the overview page.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* The link is not yet available to the public

## Testing Instructions

* Go into a WordPress.com site overview via `http://calypso.localhost:3000/sites/<site>` ensure the link points to 
* Ensure the link is pointing to `http://calypso.localhost:3000/activity-log/<site>`
<img width="2418" height="2384" alt="CleanShot 2025-09-17 at 18 35 41@2x" src="https://github.com/user-attachments/assets/f989041d-5b8b-4be5-9111-e5123dde4fda" />

* Now go into the v2 version of it `http://calypso.localhost:3000/v2/sites/<site>`
* Ensure the link is pointing to `http://calypso.localhost:3000/v2/sites/<site>/logs/activity` 
<img width="2798" height="2236" alt="CleanShot 2025-09-17 at 18 36 36@2x" src="https://github.com/user-attachments/assets/6a95a6a3-d6c1-4e1b-aa56-59defea7da49" />

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

andrea-sdl committed:
Add useUpdatedActivityLogLink prop to LatestActivityCard and SiteOverview components to redirect to proper v2 route only when needed

taipeicoder reviewed (commented):
Thanks for this fix! I would recommend to keep it simple and use window?.location?.pathname?.startsWith( '/v2' ), and set the URL accordingly.

While it's a more fragile approach, it's been used a couple of times throughout the HD code, so it'd be easy to do a search & replace when we want to remove the logic. Also it prevents props drillings 🙂

fushar commented:
> I would recommend to keep it simple and use window?.location?.pathname?.startsWith( '/v2' ), and set the URL accordingly.

Agree.

arthur791004 commented:
Yep, it’s better to just inline `window?.location?.pathname?.startsWith('/v2')` for now. We can update all the paths later once we finalize the v2 path prefix 🙂

arthur791004 commented on client/dashboard/sites/overview-latest-activity-card/index.tsx:
```diff
@@ -44,18 +44,25 @@ const fields: Field< ActivityLogEntry >[] = [
+		return `https://cloud.jetpack.com/activity-log/${ site.slug }`;
+	}
+	if ( useUpdatedActivityLogLink ) {
+		return `/v2/sites/${ site.slug }/logs/activity`;
```
Should we remove `/v2` as it causes the full page reload? Also, we need to replace `SummaryButton` with `RouterLinkSummaryButton` inside `SummaryButtonCardFooter`

andrea-sdl replied:
Implemented in https://github.com/Automattic/wp-calypso/pull/105813/commits/afb19007c8448eaee1a2c7f7bf9da129b22be8b1 and https://github.com/Automattic/wp-calypso/pull/105813/commits/3e1e305d1208b70497637d4c75b3a1c5c7e86015

Thank you for raising this!

andrea-sdl committed:
Revert "Add useUpdatedActivityLogLink prop to LatestActivityCard and SiteOverview components to redirect to proper v2 route only when needed"

This reverts commit 7aba212ae8989f31ba0198500a5dea8cf9538a2d.

andrea-sdl committed:
use windo.location to determine if we're on v2 and do the redirect.

andrea-sdl committed:
use the RouterLinkSummary button instead of the SummaryButton to not have a full page refresh

andrea-sdl committed:
remove the not needed v2 in the URL since it's handldd by the RouterLinkSummaryButton

andrea-sdl commented:
> Thanks for this fix! I would recommend to keep it simple and use window?.location?.pathname?.startsWith( '/v2' ), and set the URL accordingly.

Thanks @taipeicoder ! I reverted the changes and implemented  them again with the `startsWith` approach. I've also replaced the `SummaryButton` with the `RouterLinkSummaryButton` and removed the `v2` since it's handled by the `RouterLink`, as indicated by @arthur791004.

arthur791004 reviewed (approved):
Tested and looks good to me! The minor issue is the content will be blank when you click the button but it seems ok to me.

andrea-sdl committed:
Merge branch 'trunk' into fix/revert-activity-log-link

andrea-sdl committed:
Merge branch 'trunk' into fix/revert-activity-log-link
